### ISSUE TITLE
EiOpt is required for certification

### ISSUE THREAD

**Issue by ccann** [OpenLEADR/openleadr-python#128]
The OpenADR Alliance requires support for the following services for a `VEN - Full Function`:

- `EiEvent`
- `EiOpt`
- `EiReport`
- `RiRegisterParty`
- `OadrPoll`

I thought this might be pertinent information given that many will probably find this library in an effort to build certified VENs, and I saw in another issue that `EiOpt` is not supported and is de-prioritized. I think this is the only required service not implemented here. Please correct me if I'm misinterpreting their requirements or your documentation.

Just for clarity, I'm not asking you to do this work or complaining that it's not implemented -- just an FYI :-)

**Comment by stan-janssen** [comment]
Yes I know, just haven't gotten to implementing EiOpt. I'm very much open to pull requests on this issue!

**Comment by LinIChun** [comment]
Hi. I tried to send an oadrCreateOpt to VTN and had trouble putting dtstart and duration into the XML. They did not show up in the components of XML . My code is as shown below. Could it be an error in the format of message_payload in the code ? 
`async def create_opt_test(self):
        """
        create opt_in message to the VTN.
        """
        service = 'EiOpt'
        message_type = 'oadrCreateOpt'
        opt_id = str(utils.generate_id() + '-IN')
        opt_type = 'optIn'
        opt_reason = 'economic'
        created_date_time = datetime.now()
        request_id = utils.generate_id()
        message_payload = {
                            'components':{
                                'available':{
                                    'properties':{
                                        'dtstart': {'date-time':datetime(2023,10,19,20,00)},'duration': {'duration':timedelta(hours=1)}
                                        }}}}
        message = self._create_message(message_type,
                                        opt_id=opt_id,
                                        opt_type=opt_type,
                                        opt_reason=opt_reason,
                                        ven_id=self.ven_id,
                                        created_date_time=created_date_time,
                                        request_id=request_id,
                                        **message_payload
                                        )
    
        response_type, response_payload, vtn_xml  = await self._perform_request(service, message)
        return response_type, response_payload, vtn_xml`
